traits/trait_list_object.py

```diff
@@ -53,460 +94,754 @@ def __init__(self, index=0, removed=None, added=None):
         self.added = added
 
 
-class TraitListObject(list):
-    """ A subclass of list that fires trait events when mutated.
-
-    This is used by the List trait type, and all values set into a List
-    trait will be copied into a new TraitListObject instance.
-
-    Mutation of the TraitListObject will fire a "name_items" event with
-    appropriate index, added and removed values.  In the case of setting
-    or deleting items with a slice, the index will hold either:
-
-    - the location of the first element changed, if the step is 1.  In
-      this case the last element changed can be inferred from the
-      length of the removed values.
-    - the slice that was used, otherwise.
+class TraitList(list):
+    """ A subclass of list that validates and notifies listeners of changes.
 
     Parameters
     ----------
-    trait : CTrait instance
-        The CTrait instance associated with the attribute that this list
-        has been set to.
-    object : HasTraits instance
-        The HasTraits instance that the list has been set as an attribute for.
-    name : str
-        The name of the attribute on the object.
     value : list
-        The list of values to initialize the TraitListObject with.
+        The value for the list
+    validator : callable
+        Called to validate items in the list
+    notifiers : list of callable
+        A list of callables with the signature::
+
+         notifier(trait_list, index, removed, added)
 
     Attributes
     ----------
-    trait : CTrait instance
-        The CTrait instance associated with the attribute that this list
-        has been set to.
-    object : weak reference to a HasTraits instance
-        A weak reference to a HasTraits instance that the list has been set
-        as an attribute for.
-    name : str
-        The name of the attribute on the object.
-    name_items : str
-        The name of the items event trait that the trait list will fire when
-        mutated.
+    value : list
```

Comment:
Do we need the duplication between `Parameters` and `Attributes` here? What does NumPy documentation do in cases like this?

Reply:
> Do we need the duplication between `Parameters` and `Attributes` here? What does NumPy documentation do in cases like this?

Its uncommon, but I did see a duplication in one instance `numpy._pytesttester.PytestTester`